test/unit/datasets/test_datasets.py

```diff
@@ -38,3 +41,49 @@ def test_dataset_as_X_y(self, as_frame, fetch_function):
         assert isinstance(X, pd.DataFrame if as_frame else np.ndarray)
         assert y is not None
         assert isinstance(y, pd.Series if as_frame else np.ndarray)
+
+    def test_simple_make_sensitive_classification(self):
+        """Ensure that dataset creation is deterministic."""
+        rng = np.random.RandomState(54321)
+        gender_feature = SensitiveFeature(
+            'Gender', ['Man', 'Non-binary', 'Other', 'Unspecified', 'Woman'])
+        dataset = SensitiveDatasetMaker(sensitive_features=[gender_feature],
+                                        random_state=rng)
+        X, y, features = dataset.make_sensitive_classification(n_samples=500)
+        gender = features['Gender']
+
+        counts = [127, 370, 324, 210]
+        for i in range(4):
+            assert np.sum(y[i*500:(i+1)*500]) == counts[i]
+
+        assert np.isclose(np.mean(X), 0.0097618)
+        assert X.shape == (2500, 20)
+        assert y.shape == (2500,)
+        assert gender.shape == (2500,)
+
+    def test_custom_make_sensitive_classification(self):
```

Comment:
Would be awesome to have a test case with at least 2 sensitive features, too. In the current setup you seem to allow 0 as well, but if that's still changing we don't need a test since it would be obsolete. If you see 0 as a legitimate use case we'll need a test, too, of course.